pkg/cmd/pulumi/stack_output.go

```diff
@@ -160,6 +172,32 @@ func (*jsonStackOutputWriter) WriteMany(outputs map[string]interface{}) error {
 	return printJSON(outputs)
 }
 
+// shellStackOutputWriter prints stack outputs as a shell script.
+type shellStackOutputWriter struct {
+	W io.Writer
+}
+
+var _ stackOutputWriter = (*shellStackOutputWriter)(nil)
+
+func (w *shellStackOutputWriter) WriteOne(v interface{}) error {
+	_, err := fmt.Fprintln(w.W, stringifyOutput(v))
+	return err
+}
+
+func (w *shellStackOutputWriter) WriteMany(outputs map[string]interface{}) error {
+	keys := make([]string, 0, len(outputs))
+	for v := range outputs {
+		keys = append(keys, v)
+	}
+	sort.Strings(keys)
+
+	for _, k := range keys {
+		v := shellquote.Join(stringifyOutput(outputs[k]))
+		fmt.Fprintf(w.W, "%v=%v\n", k, v)
```

Comment:
Should we also support pwsh? Maybe just a check which platform were on and if linux/osx emit X=Y style, and if Windows assume pwsh (because who uses cmd anymore) and emit `$env:X = Y` style?

Reply:
Ack. Will investigate quoting rules to get proper escaping for `Y`.

One behavioral note: it seems that `$env:X = Y` sets the environment variable in that it is immediately available to spawned processes. Bash's `X=Y` does not do that unless you also do `export X=Y` at assignment time or `export X` afterwards. The pwsh equivalent to just `X=Y` seems to be [Set-Variable](https://learn.microsoft.com/en-us/powershell/module/microsoft.powershell.utility/set-variable?source=recommendations&view=powershell-7.3) unless I'm mistaken. So we'd have to pick one or the other—export or local variables.

On shell detection: It makes sense to pick automatically based on OS, however there should be an escape hatch.
Not in this PR, but maybe in the future, we'll want to extend the `--shell` flag to accept an optional shell type (e.g. `--shell=powershell`, `--shell=bash`) to let users pick explicitly, with `--shell` being equivalent to `--shell=auto`, which will provide the current automatic behavior. I'll create a ticket for that once I figure out the Powershell end of things.

Reply:
> The pwsh equivalent to just X=Y seems to be [Set-Variable](https://learn.microsoft.com/en-us/powershell/module/microsoft.powershell.utility/set-variable?source=recommendations&view=powershell-7.3) unless I'm mistaken.

Ah yes good point. I'm not sure what's more idiomatic for powershell. But if we are just setting variables you can use `$X = Y` syntax rather than having to emit Set-Variable calls.

Reply:
Powershell support added